src/WorkflowManager/Storage/Services/DicomService.cs

```diff
@@ -87,7 +87,9 @@ public async Task<PatientDetails> GetPayloadPatientDetailsAsync(string payloadId
                     var stream = await _storageService.GetObjectAsync(bucketId, item.FilePath);
                     var jsonStr = Encoding.UTF8.GetString(((MemoryStream)stream).ToArray());
 
-                    var dict = JsonConvert.DeserializeObject<Dictionary<string, DicomValue>>(jsonStr);
+                    var dict = new Dictionary<string, DicomValue>(StringComparer.OrdinalIgnoreCase);
```

Comment:
would it be worth implementing the sonar suggestion above?

we are only using filePath property on items so you can do...
foreach (var filePath in items.Select(item => item.FilePath)))

Reply:
Done.